vllm/entrypoints/openai/serving_engine.py

```diff
@@ -96,27 +96,32 @@ def _create_logprobs(
         last_token_len = 0
         if num_output_top_logprobs:
             logprobs.top_logprobs = []
+
         for i, token_id in enumerate(token_ids):
-            step_top_logprobs = top_logprobs[i]
-            if step_top_logprobs is not None:
-                token_logprob = step_top_logprobs[token_id].logprob
+            if top_logprobs is None or top_logprobs[i] is None:
+                token = self.tokenizer.decode(token_id)
+                logprobs.tokens.append(token)
+                logprobs.token_logprobs.append(None)
+                logprobs.top_logprobs.append(None)
```

Comment:
can you explain when would this be the case and why is the decode needed?

Reply:
This is the case when `echo == True and logprobs > 0` for the first prompt token because there is no sampling metadata (`top_logprobs[i]`) about it. The decode is needed to add the token to the logprob token list because the sampling metadata normally has it.

Reply:
The `top_logprobs is None` was actually never possible so I removed it